gammapy/data/utils.py

```diff
@@ -151,3 +154,36 @@ def get_irfs_features(
         features.meta["frame"] = "galactic"
 
     return features
+
+
+def check_time_intervals(time_intervals):
```

Comment:
This function is not working with the test as designed. Is it really needed?

Isn't the mechanism used for lightcurve, i.e. with `GTI.from_time_intervals` adapted here?

Reply:
I would also assume that a list of `Time` object (of length 2) is closer to what one would expect.

Reply:
I have fixed a logic error in `check_time_intervals` (for a strange reason, it was not appearing in my local branch, I had to independently clone the branch somewhere else).
Otherwise, `GIT.from_time_intervals` does no check at all. This is why a new function is needed.

The latest CI fails are unrelated....